smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/CommandGenerator.java

```diff
@@ -169,7 +171,13 @@ private void addInputAndOutputTypes() {
 
     private void writeInputType(String typeName, Optional<StructureShape> inputShape) {
         if (inputShape.isPresent()) {
-            writer.write("export type $L = $T;", typeName, symbolProvider.toSymbol(inputShape.get()));
+            StructureShape input = inputShape.get();
+            List<MemberShape> streamingMembers = getStramingMembers(input);
+            if (!streamingMembers.isEmpty()) {
```

Comment:
Nit: can this conditional be flipped? Streaming members are less likely to exist as well.